src/sql/mir.rs

```diff
@@ -761,6 +761,28 @@ impl SqlToMirConverter {
                 }
             }
 
+
+            // 3. Get columns used by each filter node.
+            let mut filter_columns = HashMap::new();
+            let mut filter_nodes = Vec::new();
+            let mut moved_filters = Vec::new();
+            let mut sorted_rels: Vec<&String> = qg.relations.keys().collect();
+            sorted_rels.sort();
+            for rel in &sorted_rels {
+                let qgn = &qg.relations[*rel];
+                if *rel != "computed_columns" {
+                    if !qgn.predicates.is_empty() {
```

Comment:
this is the same code that was being used to create filter nodes. the same line comes with this comment https://github.com/mit-pdos/distributary/blob/447d8a8538ef510ecd8ed98c9805df990ac13567/src/sql/mir.rs#L900 not sure if it is still a valid concern